registry/repository.go

```diff
@@ -84,6 +84,18 @@ func (c *repositoryClient) FetchSignatureBlob(ctx context.Context, desc ocispec.
 // linked signature envelope blob. Upon successful, PushSignature returns
 // signature envelope blob and manifest descriptors.
 func (c *repositoryClient) PushSignature(ctx context.Context, mediaType string, blob []byte, subject ocispec.Descriptor, annotations map[string]string) (blobDesc, manifestDesc ocispec.Descriptor, err error) {
+	// when uploading OCI artifact manifest, Notation requires the registry
+	// to support the Referrers API as well.
+	// Reference: https://github.com/opencontainers/distribution-spec/blob/v1.1.0-rc1/spec.md#listing-referrers
+	if !c.OCIImageManifest {
+		err := c.Repository.Referrers(ctx, subject, "", func(referrers []ocispec.Descriptor) error {
+			return nil
+		})
+		if err != nil {
+			return ocispec.Descriptor{}, ocispec.Descriptor{}, fmt.Errorf("failed to ping Referrers API on uploading OCI artifact manifest with error: %v. Try OCI image manifest instead", err)
+		}
+	}
```

Comment:
I'm actually thinking if this logic should be in `notation-go` or in the `notation CLI`.

Reply:
As we discussed, we still will have this overhead if we move this logic to the CLI side. Hence, I'll keep it here for now.